fcrepo/fcrepo
# fix db retries on startup; fix mysql search deadlock issues

**JIRA Ticket**: https://fedora-repository.atlassian.net/browse/FCREPO-3720

# What does this Pull Request do?

1. Fixes the the repository initialization so that it runs after the retry and scheduling infrastructure has been initialized. This is necessary so that the application does not hang waiting for an exception to be retried.
2. Fixes the search index so that it does not deadlock as much in MySQL. To do this, rdf types are added _outside_ of the current transaction.

# How should this be tested?

Fedora should start successfully when using MySQL and initialized on top of migrated content. Test data set should contain at least 100 resources and the reindexer should use multiple threads.

# Interested parties
@fcrepo/committers
